lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -303,122 +303,45 @@ export class UserAgentApplication {
     }
   }
 
-  //#endregion
+  private successHandler(interactionType: InteractionType, response: AuthResponse, resolve?: any) : void {
+    if (interactionType === Constants.interactionTypeRedirect) {
+      if (this.errorReceivedCallback) {
+        this.tokenReceivedCallback(response);
+      } else if (this.authResponseCallback) {
+        this.authResponseCallback(null, response);
+      }
+    } else if (interactionType === Constants.interactionTypePopup) {
+      resolve(response);
+    } else {
+      throw ClientAuthError.createInvalidInteractionTypeError();
+    }
+  }
 
-  //#region Redirect Flow
+  private errorHandler(interactionType: InteractionType, authErr: AuthError, response: AuthResponse, reject?: any) : void {
```

Comment:
How much of these things can be factored out into their own file? 
You have done the work to separate them, could they also be separated from state?

Reply:
A lot of this work is done in later PRs

Reply:
`reject` is a function, correct?

Reply:
correct